vue/sbc-common-components/src/components/SbcHeader.vue

```diff
@@ -107,44 +108,90 @@
+import { Component, Prop, Watch } from 'vue-property-decorator'
 import Vue from 'vue'
 import { integer } from 'vuelidate/lib/validators'
+import { initialize, LDClient } from 'launchdarkly-js-client-sdk'
+import ConfigHelper from '../util/config-helper'
+import { SessionStorageKeys } from '../util/constants'
 
 @Component({})
 export default class SbcHeader extends Vue {
   @Prop({ default: '-' }) private accountName: string;
   @Prop({ default: 0 }) private pendingApprovalCount: number;
-  get username () : string {
-    return sessionStorage.getItem('USER_FULL_NAME') || '-'
+  private ldClient: LDClient;
+
+  get showAccountSwitching (): boolean {
+    try {
+      const flags = JSON.parse(ConfigHelper.getFromSession(SessionStorageKeys.LaunchDarklyFlags))
+      return flags && flags['account-switching']
+    } catch (exception) {
+      return false
+    }
+  }
+
+  private get _accountName (): string {
+    return this.accountName || ConfigHelper.getFromSession(SessionStorageKeys.AccountName) || '-'
+  }
+
+  private get _pendingApprovalCount (): number {
+    try {
+      return this.pendingApprovalCount ? this.pendingApprovalCount : parseInt(ConfigHelper.getFromSession(SessionStorageKeys.PendingApprovalCount) || '0')
+    } catch (exception) {
+      return 0
+    }
+  }
+
+  @Watch('accountName')
+  private onAccountNameChanged (val: string, oldVal: string): void {
+    ConfigHelper.addToSession(SessionStorageKeys.AccountName, val)
+  }
+
+  @Watch('pendingApprovalCount')
+  private onPendingApprovalCountChanged (val: number, oldVal: number): void {
+    ConfigHelper.addToSession(SessionStorageKeys.PendingApprovalCount, val)
   }
 
-  get authorized (): boolean {
-    let auth = sessionStorage.getItem('KEYCLOAK_TOKEN')
+  private get username () : string {
+    return ConfigHelper.getFromSession(SessionStorageKeys.UserFullName) || '-'
+  }
+
+  private get authorized (): boolean {
+    let auth = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken)
     return !!auth
   }
 
-  get accountType (): string {
-    return sessionStorage.getItem('USER_ACCOUNT_TYPE') || 'BCSC'
+  private get accountType (): string {
+    return ConfigHelper.getFromSession(SessionStorageKeys.UserAccountType) || 'BCSC'
+  }
+
+  private mounted () {
+    // Initialize LaunchDarkly flags and sync to session storage
+    const user = { 'key': 'sbc-common-components' }
+    this.ldClient = initialize('5db9da115f58e008123cd783', user)
+    this.ldClient.on('ready', () => {
+      console.log('It\'s now safe to request feature flags')
+      ConfigHelper.addToSession(SessionStorageKeys.LaunchDarklyFlags, JSON.stringify(this.ldClient.allFlags()))
+    })
   }
 
-  logout () {
+  private logout () {
     window.location.assign('/cooperatives/auth/signout')
   }
 
-  login () {
+  private login () {
     window.location.assign('/cooperatives/auth/signin/bcsc')
   }
 
-  goToUserProfile () {
+  private goToUserProfile () {
```

Comment:
Not sure what this comment has to do with marking the routine as private.  The intent is to prevent the routines from called from outside the component.

Reply:
Nothing with making this private. Sorry, was just curious.